src/core/maprenderer/qgsmaprendererjob.h

```diff
@@ -490,6 +490,14 @@ class CORE_EXPORT QgsMapRendererJob : public QObject
 
     //! Convenient method to allocate a new image and a new QPainter on this image
     QPainter *allocateImageAndPainter( QString layerId, QImage *&image );
+
+    /**
+     *  This virtual method has to be implemented in derived class for starting the rendering.
+     *  This method is called in start() method after ckecking if the map can be rendered.
+     *  \since QGIS 3.20
+     */
+    virtual void startPrivate() {};
```

Comment:
```suggestion
    virtual void startPrivate() = 0;
```

Reply:
While this is technically an API break I would argue that QgsMapRendererJob was never designed to be subclassed by Python, and any plugins/scripts which ARE subclassing it are an extreme edge case which falls firmly into the "not officially supported" camp!

Reply:
This method and this doc is in the private part of the class, so not exposed to Python, no? Or I have misunderstoodd your comment...

For the pure virtual method, my first intention was to make this method pure virtual, but I couldn't make SIP accept it: when building, I've got some error with the sub classes (I don't remember exactly which errors...), the only to make the building was to not make this method not pure virtual.

Reply:
You might need to add SIP_ABSTRACT to the base class definition

e.g. 

```
class CORE_EXPORT QgsMapLayerRenderer SIP_ABSTRACT
```

Reply:
Indeed, much better, and also for abstract sub classes.